sources/isal/igzip/igzip.c

```diff
@@ -689,7 +689,7 @@ static void qpl_isal_deflate_int(struct isal_zstream* stream, uint8_t * start_in
         state->tmp_out_start += size;
 
         if (state->tmp_out_start == state->tmp_out_end)
-            state->state -= ZSTATE_TMP_OFFSET;
+            state->state = static_cast<isal_zstate_state>(state->state - ZSTATE_TMP_OFFSET);
```

Comment:
C++ compilers are less permissive about mixing enum and integral values. These C++-style casts were automatically generated using a large language model.